### PR TITLE
TST: Move some Series ctor tests to SharedWithSparse

### PR DESCRIPTION
**Pull request by kernc**
- [x] closes https://github.com/pandas-dev/pandas/pull/16906
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [x] whatsnew entry

Continues https://github.com/pandas-dev/pandas/pull/16906.

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -282,7 +282,7 @@ Groupby/Resample/Rolling
 
 Sparse
 ^^^^^^
-- Bug in ``SparseSeries`` raises ``AttributeError`` when a dictionary is passed in as data (:issue:`16777`)
+- Bug in ``SparseSeries`` raises ``AttributeError`` when a dictionary is passed in as data (:issue:`16905`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: oh typo from before?

File: pandas/tests/series/test_api.py [hunk 2]
```diff
@@ -21,6 +22,12 @@
 
 class SharedWithSparse(object):
 
+    def _assert_series_equal(self, left, right):
```
Comment (jreback) [inline pandas/tests/series/test_api.py]: why don't we instead define this in the Mixin classes? (and add a NotImplemented one here)?

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (kernc) [inline doc/source/whatsnew/v0.21.0.txt]: Yea, https://github.com/pandas-dev/pandas/pull/16960

File: pandas/tests/sparse/test_series.py [hunk 3]
```diff
@@ -58,6 +60,9 @@ def _test_data2_zero():
 
 class TestSparseSeries(SharedWithSparse):
 
+    series_klass = SparseSeries
+    _assert_series_equal = staticmethod(tm.assert_sp_series_equal)
```
Comment (kernc) [inline pandas/tests/sparse/test_series.py]: This doesn't look as good, but avoids `*args, **kwargs` passing and retains docstring.

Comment (jreback) [inline pandas/tests/sparse/test_series.py]: hmm, I like that actually.

The only thing is we are using ``self._assert_series_equal``, which is a tiny bit annoying, when in the rest of the codebase we are using ``tm.assert_series_equal`` (or ``tm.assert_sp_series_equal``.

any ideas on a better / more consistent way would be good. (maybe add a comment to this effect in the sub-classes about this)

Comment (kernc) [inline pandas/tests/sparse/test_series.py]: In `setup_method()`:
```py
pd.util.testing.assert_series_equal = whatever
```
J/k. I have nothing better.

**Comment by jreback** [comment]
thanks @kernc 

happy to have you do the same for SparseFrame as well.

**Pull request merged**